lib/parsers/index.ts

```diff
@@ -52,9 +52,15 @@ export interface LockfileParser {
 
 export type Lockfile = PackageLock | YarnLock;
 
-export function parseManifestFile(manifestFileContents: string): ManifestFile {
+export function parseManifestFile(manifestFileContents: string, manifestFileName: string): ManifestFile {
   try {
-    return JSON.parse(manifestFileContents);
+    const manifestFile = JSON.parse(manifestFileContents);
+    // Fallback to the file name if name is not set
+    if (!manifestFile.name) {
+        manifestFile.name = manifestFileName;
```

Comment:
Can I suggest to move all logic relevant to `manifestFileName` to `/lib/index.ts`? Looks like it is not related to parsers because they only need to parse existing file. And also it is less changes (do not need to pass `manifestFileName` everywhere).